From 6c14772e9ee3af2e7216f21fc7a11dba0bc315f8 Mon Sep 17 00:00:00 2001
From: Eric Huang <JinHuiEric.Huang@amd.com>
Date: Fri, 10 Feb 2017 14:33:19 -0500
Subject: [PATCH 1945/2094] drm/amdgpu: fix a bug for automatically setting
 compute power profile

It is for enabling automatic switch on CI(Hawii) which has no powerplay.

Change-Id: Ifbe1fe6aad8eee4d31e58a41d450af58a74d8dc0
Signed-off-by: Eric Huang <JinHuiEric.Huang@amd.com>
---
 drivers/gpu/drm/amd/amdgpu/amdgpu_vm.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/amd/amdgpu/amdgpu_vm.c b/drivers/gpu/drm/amd/amdgpu/amdgpu_vm.c
index 7cc01c6..e92ae53 100644
--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_vm.c
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_vm.c
@@ -2347,7 +2347,7 @@ int amdgpu_vm_init(struct amdgpu_device *adev, struct amdgpu_vm *vm,
 	amdgpu_bo_unreserve(vm->root.bo);
 
 	vm->is_kfd_vm = is_kfd_vm;
-	if (is_kfd_vm && adev->pp_enabled) {
+	if (is_kfd_vm) {
 		mutex_lock(&adev->vm_manager.lock);
 
 		if (adev->vm_manager.n_kfd_vms++ == 0) {
@@ -2413,7 +2413,7 @@ void amdgpu_vm_fini(struct amdgpu_device *adev, struct amdgpu_vm *vm)
 	bool prt_fini_needed = !!adev->gart.gart_funcs->set_prt;
 	int i;
 
-	if (vm->is_kfd_vm && adev->pp_enabled) {
+	if (vm->is_kfd_vm) {
 		mutex_lock(&adev->vm_manager.lock);
 
 		WARN(adev->vm_manager.n_kfd_vms == 0, "Unbalanced number of KFD VMs");
-- 
2.7.4

